hack4impact-calpoly/five-cities-food-pantry
# Flagging half done & Linking clients to own page

## Developer: Hope Yim

Closes #50 and #62

### Pull Request Summary

The purpose of this PR is to add the flag modal and icon to the corresponding pages. Currently due to time constraints, I finished the first part of the task and had to make some modifications to the original designs (I asked the designer for approval in which she liked). In this task, I was able to display the clients on the home page who are flagged or not and show the icon in their own client page if they are flagged. Temporarily, I added a button for those who are NOT flagged 'Flag Client?' as a way for the user to add text in the modal for flagging reasons. 

I also linked the client's name to their corresponding page and added a back link to the home page (the  <- Back link)

### Modifications
- route.ts in flag/userId clients
- clientDashboard.tsx and .css
- clientInformationTable.tsx
- tableRow.tsx and .css
- clientSchema.ts

### Testing Considerations

- Please check Alice or David to see their client page to see if they are flagged or not and the correct icon/button shows up based on the boolean value isFlagged
- Check the home page to see if it follows the database.
- The modal is not functional so the API PUT hook is wrong on the confirm button but will be done soon.
- Click on the client's name and navigate to their page. Check the modals and click on the Back button to go back.

### Pull Request Checklist


### Screenshots/Screencast

<img width="1275" alt="Screen Shot 2024-05-29 at 3 16 54 PM" src="https://github.com/hack4impact-calpoly/five-cities-food-pantry/assets/47905779/744e42b2-9803-4603-90ff-8994a5026466">
<img width="1173" alt="Screen Shot 2024-05-29 at 3 17 13 PM" src="https://github.com/hack4impact-calpoly/five-cities-food-pantry/assets/47905779/9f30319b-952d-4c31-9fb3-6d81c345416c">
<img width="1165" alt="Screen Shot 2024-05-29 at 3 17 42 PM" src="https://github.com/hack4impact-calpoly/five-cities-food-pantry/assets/47905779/db293273-2ddd-4d4b-8b5b-c30df42d8d40">
